grails-app/migrations/views/pick-task.sql

```diff
@@ -26,21 +26,28 @@ SELECT
     pli.bin_location_id AS location_id,
     pli.outbound_container_id AS outbound_container_id,
     (SELECT loc.id
-         FROM location loc
-         JOIN location_effective_supported_activities act ON act.location_id = loc.id
-         WHERE loc.parent_location_id = r.origin_id
-           AND loc.active = true
-           AND act.supported_activities_string =
-               CASE r.delivery_type_code
-                   WHEN 'PICK_UP' THEN 'DELIVERY_TYPE_PICKUP'
-                   WHEN 'LOCAL_DELIVERY' THEN 'DELIVERY_TYPE_LOCAL_DELIVERY'
-                   WHEN 'SERVICE' THEN 'DELIVERY_TYPE_SERVICE'
-                   WHEN 'WILL_CALL' THEN 'DELIVERY_TYPE_WILL_CALL'
-                   WHEN 'SHIP_TO' THEN 'DELIVERY_TYPE_SHIPPING'
-                   ELSE NULL
-               END
-         LIMIT 1
-        ) AS staging_location_id,
+     FROM location loc
+     JOIN location_type lt ON lt.id = loc.location_type_id
+     JOIN location_effective_supported_activities lesa ON lesa.location_id = loc.id
+     WHERE loc.parent_location_id = r.origin_id
+       AND loc.active = true
+       AND lt.name = 'Cross-docking'
```

Comment:
Can you explain the need for this?

Reply:
Yeah, as we discussed on Slac,k we should have a new activity code, or a specific set of existing activity codes (i am not sure if we have good candidates right now, so it would rather be a new one)